contracts/src/Teleporter/tests/GetRelayerRewardAddressTests.t.sol

```diff
@@ -16,26 +16,43 @@ contract GetRelayerRewardAddressTest is TeleporterMessengerTest {
     }
 
     function testSuccess() public {
-        // Before receiving the message, it returns the 0 address.
-        uint256 mockMessageID = 8;
-        assertEq(
-            teleporterMessenger.getRelayerRewardAddress(DEFAULT_DESTINATION_CHAIN_ID, mockMessageID),
-            address(0)
+        // Receive a message
+        uint256 mockNonce = 8;
+        bytes32 mockMessageID = teleporterMessenger.calculateMessageID(
+            DEFAULT_ORIGIN_BLOCKCHAIN_ID, DEFAULT_DESTINATION_BLOCKCHAIN_ID, mockNonce
         );
-
-        // Mock receiving the message
         address relayerRewardAddress = 0xCAFebAbeDc0D4D7B7EEdCf61eb863fF413BB6234;
         _receiveTestMessage(
-            DEFAULT_DESTINATION_CHAIN_ID,
-            mockMessageID,
+            DEFAULT_ORIGIN_BLOCKCHAIN_ID,
+            mockNonce,
             relayerRewardAddress,
             new TeleporterMessageReceipt[](0)
         );
 
         // Now it has the relayer reward address.
-        assertEq(
-            teleporterMessenger.getRelayerRewardAddress(DEFAULT_DESTINATION_CHAIN_ID, mockMessageID),
-            relayerRewardAddress
+        assertEq(teleporterMessenger.getRelayerRewardAddress(mockMessageID), relayerRewardAddress);
+    }
+
+    function testZeroRelayerRewardAddress() public {
+        // Receive a message with a zero relayer reward address
+        uint256 mockNonce = 4343;
```

Comment:
I'm not sure how the VM state is handled under the hood, or if it persists from test to test. But if these tests rely on the mockNonces in the individual tests to be unique, then it seems pretty error prone. If that's the case, we should consider adding a utility global to all the tests to ensure uniqueness.

Reply:
Each individual test gets run against a fresh contract (created by `setUp()`), so no state  is ever shared between multiple forge unit tests. Because of that, I don't think this is an issue.

Reply:
Cool, thanks for the explanation. I wasn't sure if the nonce values being different from test to test was pertinent or not.